### ISSUE TITLE
Help

### ISSUE THREAD

**Issue by heshanlk** [kalinchernev/drupal_block_reactive#1]
This is a small help, I come from the Drupal world and no very minimum about ReactJS, looks like you've done some cool stuff with it and I just want to know is it possible to render just HTML elements from ReactJS within Drupal.

I'm trying to build Audio player within Drupal using this lib but not sure if we use the same approach as the Github project will that help.

Here is the project : https://github.com/humanhighway/react-audio-player

Please advise.

**Comment by kalinchernev** [comment]
Hi @heshanlk actually I'm a Drupal guy too, playing and experimenting with react here in my weekend, i.e. that's why there's no webpack ;)
I think the audio player you want to use should be ok with any other web app, as long as you include an element with an id which [instantiates the component](https://github.com/humanhighway/react-audio-player/blob/master/js/app.js#L12).
In my example, the id is [recent-comments-react](https://github.com/kalinchernev/drupal_block_reactive/blob/master/js/src/index.jsx#L74) which is included by a block, but this is super-flexible thing - you have many ways to alter or include markup in Drupal.
So basically, the steps you need to take to implement any react component:
- build a rest api to communicate with the component (comments endpoint in this project example)
- include the necessary assets on the page: react, reactdom, etc.
- include markup with an id which instantiates the component
- include the transpiled/bundled/compiled/etc react component (in D7 [drupal_add_js or #attached](https://www.drupal.org/node/304255) and in D8 is only #attach property and libraries api)
- load the page and work with the component.

I hope this helps, closing the ticket as it's not really an issue with the project.

**Comment by heshanlk** [comment]
Thanks, that helps. I'll dig little more in to it and see how far I can go with this.
